spiketoolkit/postprocessing/postprocessing_tools.py

```diff
@@ -848,66 +892,205 @@ def _get_random_spike_waveforms(recording, sorting, unit, max_spikes_per_unit, s
     return spikes, event_indices
 
 
-def _get_quality_metric_data(recording, sorting, nPC, ms_before, ms_after, dtype, amp_method, amp_peak,
-                             amp_frames_before, amp_frames_after, max_spikes_per_unit, max_spikes_for_pca, \
-                             recompute_waveform_info, save_features_props, verbose, seed):
-    if recompute_waveform_info:
+def _get_spike_times_clusters(sorting):
+    if not isinstance(sorting, se.SortingExtractor):
+        raise AttributeError()
+    if len(sorting.get_unit_ids()) == 0:
+        raise Exception("No units in the sorting result, can't compute any metric information.")
+
+    # spike times.npy and spike clusters.npy
+    spike_times = np.array([])
+    spike_clusters = np.array([])
+
+    for i_u, unit_id in enumerate(sorting.get_unit_ids()):
+        spike_train = sorting.get_unit_spike_train(unit_id)
+        cl = [i_u] * len(sorting.get_unit_spike_train(unit_id))
+        spike_times = np.concatenate((spike_times, np.array(spike_train)))
+        spike_clusters = np.concatenate((spike_clusters, np.array(cl)))
+
+    sorting_idxs = np.argsort(spike_times)
+    spike_times = spike_times[sorting_idxs, np.newaxis]
+    spike_clusters = spike_clusters[sorting_idxs, np.newaxis].astype(int)
+
+    return spike_times, spike_clusters
+
+
+def _get_amp_metric_data(recording, sorting, amp_method, amp_peak,
+                         amp_frames_before, amp_frames_after, max_spikes_per_unit, recompute_info,
+                         save_features_props, seed):
+    if recompute_info:
+        sorting.clear_units_spike_features(feature_name='amplitudes')
+
+    # amplitudes.npy
+    amplitudes_list, amp_idxs = get_unit_amplitudes(recording, sorting, method=amp_method,
+                                                    save_as_features=save_features_props, peak=amp_peak,
+                                                    max_spikes_per_unit=max_spikes_per_unit,
+                                                    frames_before=amp_frames_before, frames_after=amp_frames_after,
+                                                    seed=seed, return_idxs=True)
+
+    # spike times.npy and spike clusters.npy
+    spike_times = np.array([])
+    spike_clusters = np.array([])
+    amplitudes = np.array([])
+
+    for i_u, id in enumerate(sorting.get_unit_ids()):
+        st = sorting.get_unit_spike_train(id)
+        amp = amplitudes_list[i_u]
+
+        if len(amp) < len(st):
+            cl = [i_u] * len(amp)
+            spike_times = np.concatenate((spike_times, st[amp_idxs[i_u]]))
```

Comment:
concatenating in a loop is a bad pratice. It is better to preallocate the good array sioze from the beggning or concatenating at the end a big list.

Reply:
I could fix that here since we now the indexes, but I think that since these are 1d array it's fast anyways